### ISSUE TITLE
Inconsistent results

### ISSUE THREAD

**Issue by sherryyin1** [Li-Muyang/MLP4Rec#1]
Hi, Thanks for your interesting works. When I trained the model, I found that it was inconsistent with the results in the paper, and the training loss was very large. The hyperparameter settings are all default, I just ran run_mlp4rec.py.

**Comment by Li-Muyang** [comment]
Hi, thanks for bringing this up! Do you mind copy the console output to this thread so that we can have a look what seems to be the problem?

**Comment by sherryyin1** [comment]
Mon 19 Sep 2022 22:47:41 INFO  epoch 0 training [time: 1447.52s, train loss: 4454.5431]
Mon 19 Sep 2022 22:58:09 INFO  epoch 0 evaluating [time: 628.51s, valid_score: 0.097000]
Mon 19 Sep 2022 22:58:09 INFO  valid result:
recall@10 : 0.2717    mrr@10 : 0.097    ndcg@10 : 0.1373    hit@10 : 0.2717    precision@10 : 0.0272
Mon 19 Sep 2022 22:58:09 INFO  Saving current best: saved/MLP4Rec-Sep-19-2022_22-23-33.pth
Mon 19 Sep 2022 23:22:14 INFO  epoch 1 training [time: 1444.15s, train loss: 3994.5049]
Mon 19 Sep 2022 23:32:42 INFO  epoch 1 evaluating [time: 628.02s, valid_score: 0.179700]
Mon 19 Sep 2022 23:32:42 INFO  valid result:
recall@10 : 0.3995    mrr@10 : 0.1797    ndcg@10 : 0.2313    hit@10 : 0.3995    precision@10 : 0.04
Mon 19 Sep 2022 23:32:42 INFO  Saving current best: saved/MLP4Rec-Sep-19-2022_22-23-33.pth
Mon 19 Sep 2022 23:56:46 INFO  epoch 2 training [time: 1444.31s, train loss: 3687.4842]
Tue 20 Sep 2022 00:07:16 INFO  epoch 2 evaluating [time: 630.31s, valid_score: 0.221700]
Tue 20 Sep 2022 00:07:16 INFO  valid result:
recall@10 : 0.4495    mrr@10 : 0.2217    ndcg@10 : 0.2753    hit@10 : 0.4495    precision@10 : 0.045
Tue 20 Sep 2022 00:07:17 INFO  Saving current best: saved/MLP4Rec-Sep-19-2022_22-23-33.pth
Tue 20 Sep 2022 00:31:22 INFO  epoch 3 training [time: 1445.17s, train loss: 3423.5285]
Tue 20 Sep 2022 00:41:46 INFO  epoch 3 evaluating [time: 623.91s, valid_score: 0.234500]
Tue 20 Sep 2022 00:41:46 INFO  valid result:
recall@10 : 0.4621    mrr@10 : 0.2345    ndcg@10 : 0.2882    hit@10 : 0.4621    precision@10 : 0.0462
Tue 20 Sep 2022 00:41:46 INFO  Saving current best: saved/MLP4Rec-Sep-19-2022_22-23-33.pth
Tue 20 Sep 2022 01:05:49 INFO  epoch 4 training [time: 1443.04s, train loss: 3220.5126]
Tue 20 Sep 2022 01:16:13 INFO  epoch 4 evaluating [time: 624.33s, valid_score: 0.239300]

n_layers = 4
hidden_size = 128
hidden_dropout_prob = 0.5
hidden_act = gelu
layer_norm_eps = 1e-12
initializer_range = 0.02
selected_features = ['sales_rank', 'categories', 'price', 'brand']
pooling_mode = mean
loss_type = CE
MODEL_TYPE = ModelType.SEQUENTIAL
MODEL_INPUT_TYPE = InputType.POINTWISE
eval_type = EvaluatorType.RANKING
device = cuda
train_neg_sample_args = {'strategy': 'none'}
eval_neg_sample_args = {'strategy': 'by', 'by': 100, 'distribution': 'popularity'}

I am confused and looking forward to your reply.

**Comment by Li-Muyang** [comment]
Hi, the result seems to be in an acceptable range, which metric exactly do you think is inconsistent?

**Comment by sherryyin1** [comment]
I'm a beginner for the recommendation task, just I think the training loss seems a bit big.

**Comment by Li-Muyang** [comment]
Hi, based on my experience, I would say this training loss is quite common with respect to this particular dataset. Once the model is converge, you should observe a similar result as our paper presented :)
I hope this answers your question, if you have any further question please free to contact me or open a new issue anytime.

**Comment by sherryyin1** [comment]
Thanks for your reply, I get it !

**Comment by Li-Muyang** [comment]
Hi, thanks for bringing this up! Can you copy the console output into a
text file and send that to me?


sherryyin1 ***@***.***>于2022年9月19日 周一下午10:39写道：
